task/backend/executor/executor_test.go

```diff
@@ -380,92 +381,66 @@ func testMetrics(t *testing.T) {
 
 	mg := promtest.MustGather(t, reg)
 	m := promtest.MustFindMetric(t, mg, "task_executor_total_runs_active", nil)
-	if got := *m.Gauge.Value; got != 0 {
-		t.Fatalf("expected 0 total runs active, got %v", got)
-	}
+	assert.EqualValues(t, 0, *m.Gauge.Value, "unexpected number of active runs")
 
 	script := fmt.Sprintf(fmtTestScript, t.Name())
 	ctx := icontext.SetAuthorizer(context.Background(), tes.tc.Auth)
 	task, err := tes.i.CreateTask(ctx, influxdb.TaskCreate{OrganizationID: tes.tc.OrgID, OwnerID: tes.tc.Auth.GetUserID(), Flux: script})
-	if err != nil {
-		t.Fatal(err)
-	}
+	assert.NoError(t, err)
 
 	promise, err := tes.ex.PromisedExecute(ctx, scheduler.ID(task.ID), time.Unix(123, 0), time.Unix(126, 0))
-	if err != nil {
-		t.Fatal(err)
-	}
-	promiseID := influxdb.ID(promise.ID())
+	assert.NoError(t, err)
+	promiseID := promise.ID()
 
 	run, err := tes.i.FindRunByID(context.Background(), task.ID, promiseID)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if run.ID != promiseID {
-		t.Fatal("promise and run dont match")
-	}
+	assert.NoError(t, err)
+	assert.EqualValues(t, promiseID, run.ID, "promise and run dont match")
 
 	tes.svc.WaitForQueryLive(t, script)
 
 	mg = promtest.MustGather(t, reg)
 	m = promtest.MustFindMetric(t, mg, "task_executor_total_runs_active", nil)
-	if got := *m.Gauge.Value; got != 1 {
-		t.Fatalf("expected 1 total runs active, got %v", got)
-	}
+	assert.EqualValues(t, 1, *m.Gauge.Value, "unexpected number of active runs")
 
 	tes.svc.SucceedQuery(script)
 	<-promise.Done()
 
-	mg = promtest.MustGather(t, reg)
+	// N.B. You might think the _runs_complete and _runs_active metrics are updated atomically,
+	// but that's not the case. As a task run completes and is being cleaned up, there's a small
+	// window where it can be counted under both metrics.
+	//
+	// Our CI is very good at hitting this window, causing failures when we assert on the metric
+	// values below. We sleep a small amount before gathering metrics to avoid flaky errors.
```

Comment:
As seen in: https://app.circleci.com/pipelines/github/influxdata/influxdb/17163/workflows/ab6a5788-049a-4e2f-af15-793681d73b73/jobs/163208